base_tier_validation/migrations/16.0.3.1.0/pre-migration.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2025 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    env.cr.execute(
+        """
+        SELECT imf.model
+        FROM ir_model_fields AS imf
+        WHERE imf.name = 'review_ids'
+        AND imf.ttype = 'one2many'
+        AND imf.model != 'tier.validation'
+        """
+    )
+    for (model_name,) in env.cr.fetchall():
+        table_name = model_name.replace(".", "_")
+        # validated column
+        if not openupgrade.column_exists(env.cr, table_name, "validated"):
+            openupgrade.logged_query(
+                env.cr,
+                f"""
+                ALTER TABLE {table_name}
+                ADD COLUMN IF NOT EXISTS validated BOOLEAN
+                """,
+            )
+            # Define all those with reviews as validated and then define
+            # as not validated those with unapproved reviews.
+            openupgrade.logged_query(
+                env.cr,
+                f"""
+                UPDATE {table_name} SET validated = true WHERE id IN (
+                    SELECT DISTINCT(tr.res_id)
+                    FROM tier_review AS tr
+                    WHERE tr.model = '{model_name}'
+                )
+                """,
+            )
+            openupgrade.logged_query(
+                env.cr,
+                f"""
+                UPDATE {table_name} SET validated = false WHERE id IN (
+                    SELECT DISTINCT(tr.res_id)
+                    FROM tier_review AS tr
+                    WHERE tr.model = '{model_name}' AND tr.status != 'approved'
```

Comment:
```suggestion
                    WHERE tr.model = '{model_name}' AND tr.status not in ('approved', 'forwarded')
```
if I understand correctly.